addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -608,17 +596,24 @@ ListRenderer.include({
         }
         const table = this.el.getElementsByTagName('table')[0];
         const thead = table.getElementsByTagName('thead')[0];
+        const thElements = [...thead.getElementsByTagName('th')];
 
         // Set table layout auto and remove inline style to make sure that css
         // rules apply (e.g. fixed width of record selector)
         table.style.tableLayout = 'auto';
-        [...table.getElementsByTagName('th')].forEach((th) => {
+        thElements.forEach((th) => {
             th.style.width = null;
+            th.style.maxWidth = null;
         });
 
-        // Freeze each th width according to their size in auto layout
-        const thElements = [...thead.getElementsByTagName('th')];
-        const thWidths = thElements.map((th) => th.offsetWidth);
+        const allowedWidth = table.parentNode.offsetWidth;
+        const totalWidth = thElements.reduce((total, th) => total + th.offsetWidth, 0);
+
+        if (totalWidth > allowedWidth) {
+            this._trimLargestColumns(totalWidth, allowedWidth);
+        }
+
+        const thWidths = thElements.map(th => Number(th.style.maxWidth.split('px')[0]) || th.offsetWidth);
```

Comment:
not a fan of the maxWidth.split stuff, maybe we should try to find another way (like storing the max widths we set in another structure)